tests/test_pricing_model.py

```diff
@@ -0,0 +1,1279 @@
+                    days_remaining=91.25,  # 3 months remaining
+                    time_stretch_apy=0.05,  # APY of 5% used to calculate time_stretch
+                    share_price=2,  # share price of the LP in the yield source
+                    init_share_price=1.5,  # original share price pool started
+                ),
+                # From the input, we have the following values:
+                # T = 22.1868770168519182502689135891
+                # τ = 91.25/365/22.1868770168519182502689135891 = 0.011267922015798524
+                # 1 - τ = 0.9887320779842015
+                # k = c/u*(u*z)**(1-τ) + (2*y + c*z)**(1-τ)
+                # k = 2/1.5*((1.5*100000)**0.9887320779842015) + (2*1000000 + 2*100000)**0.9887320779842015
+                # k = 2041060.1949973335
+                (
+                    TradeResult(
+                        # p = ((2y+cz)/uz)**τ
+                        #   = ((2*1000000 + 2*100000)/(1.5*100000))**0.011267922015798524
+                        #   = 1.0307233899745727
+                        # without_fee_or_slippage = 1/p * out = 194.038480105641
+                        without_fee_or_slippage=194.038480105641,
+                        # fee is 10% of discount before slippage = (200-194.038480105641)*0.1 = 0.5961519894358986
+                        fee=0.5961519894358986,
+                        # deltaZ = 1/u * (u/c*(k - (2*y + c*z - deltaY)**(1-τ)))**(1/(1-τ)) - z
+                        # deltaZ = 2*(1/1.5 * (1.5/2*(2041060.1949973335 - (2*1000000 + 2*100000 - 200)**(1-0.011267922015798524)))**(1/(1-0.011267922015798524)) - 100000)
+                        #        = 194.0396397759323
+                        without_fee=194.0396397759323,
+                        # with_fee = without_fee + fee = 194.0396397759323 +  0.5961519894358986 = 194.6357917653682
+                        with_fee=194.6357917653682,
+                    )
+                ),
+            ),  # end of test six
+        ]
+        for (test_case, expected) in test_cases:
+            time_stretch = pricing_model.calc_time_stretch(test_case.time_stretch_apy)
+            time_remaining = stretch_time(pricing_model.days_to_time_remaining(test_case.days_remaining), time_stretch)
+
+            # Ensure we get the expected results from the pricing model.
+            (without_fee_or_slippage, with_fee, without_fee, fee) = pricing_model.calc_in_given_out(
+                test_case.out,
+                test_case.share_reserves,
+                test_case.bond_reserves,
+                test_case.token_in,
+                test_case.fee_percent,
+                time_remaining,
+                test_case.init_share_price,
+                test_case.share_price,
+            )
+            actual = TradeResult(without_fee_or_slippage, with_fee, without_fee, fee)
+            compare_trade_results(actual, expected)
+
+    def test_calc_out_given_in_success(self):
```

Comment:
should probably rename this to match the other function, i.e. `test_calc_out_given_in` right?

Reply:
Oh, I see, there is a `success` and `failure` test. Then do we need those for in_given_out?

Reply:
Yes we do. @wakamex should be on it, but please let me know if you need me to handle it @wakamex.